go/nbs/table_writer.go

```diff
@@ -70,28 +68,19 @@ func newTableWriter(buff []byte, snapper snappyEncoder) *tableWriter {
 }
 
 func (tw *tableWriter) addChunk(h addr, data []byte) bool {
-	if len(data) == 0 {
-		panic("NBS blocks cannont be zero length")
-	}
+	dataLength := uint64(len(data))
 
-	// Compress data straight into tw.buff
-	compressed := tw.snapper.Encode(tw.buff[tw.pos:], data)
-	dataLength := uint64(len(compressed))
-	tw.totalCompressedData += dataLength
-
-	// BUG 3156 indicated that, sometimes, snappy decided that there's not enough space in tw.buff[tw.pos:] to encode into.
-	// This _should never happen anymore be_, because we iterate over all chunks to be added and sum the max amount of space that snappy says it might need.
-	// Since we know that |data| can't be 0-length, we also know that the compressed version of |data| has length greater than zero. The first element in a snappy-encoded blob is a Uvarint indicating how much data is present. Therefore, if there's a Uvarint-encoded 0 at tw.buff[tw.pos:], we know that snappy did not write anything there and we have a problem.
-	if v, n := binary.Uvarint(tw.buff[tw.pos:]); v == 0 {
-		d.Chk.True(n != 0)
-		panic(fmt.Errorf("BUG 3156: unbuffered chunk %s: uncompressed %d, compressed %d, snappy max %d, tw.buff %d\n", h.String(), len(data), dataLength, snappy.MaxEncodedLen(len(data)), len(tw.buff[tw.pos:])))
+	if dataLength == 0 {
+		panic("NBS blocks cannont be zero length")
 	}
 
+	copy(tw.buff[tw.pos:], data)
```

Comment:
This is a big departure from the past. We definitely found that being able to compress chunk data directly into `tw.buff` was a big win. This might be OK, but we should check and consider figuring out a way to get that win back.